File: backend/models/yeet.js
const mongoose = require('mongoose');

const yeetSchema = new mongoose.Schema({
  content: {
    type: String,
    required: true,
  },
  media: {
    type: String,
    default: '',
  },
  author: {
    type: mongoose.Schema.Types.ObjectId,
    ref: 'User',
    required: true,
  },
  group: {
    type: mongoose.Schema.Types.ObjectId,
    ref: 'Group',
  },
  timestamp: {
    type: Date,
    default: Date.now,
  },
  replyTo: {
    type: mongoose.Schema.Types.ObjectId,
    ref: 'Yeet',
  },
  thread: {
    type: mongoose.Schema.Types.ObjectId,
    ref: 'Yeet',
  },
  source: {
    type: String,
  },
  // Add more fields for media references if needed
});

const Yeet = mongoose.model('Yeet', yeetSchema);

module.exports = Yeet;

File: backend/src/main/java/com/yeeter/yeeterBackend/entities/Yeet.java
package com.yeeter.yeeterBackend.entities;

import lombok.Data;
import org.springframework.data.annotation.Id;
import org.springframework.data.mongodb.core.mapping.DBRef;
import org.springframework.data.mongodb.core.mapping.Document;

import java.util.Date;

@Data
@Document(collection = "yeet")
public class Yeet {

    @Id
    private String id;

    private String content;

    private String media = "";

    @DBRef
    private User author;

    //jj@DBRef
    //private Group group;

    private Date timestamp = new Date();

    @DBRef
    private Yeet replyTo;

    @DBRef
    private Yeet thread;

    private String source;
}
